### PR TITLE
Implement buttons states

### LINKED ISSUE #47: Implement buttons states
- Choose polling station screen: 'continue' button should be enabled only if the county and polling station number are filled in
- Polling station details screen: 'continue to forms' button should be enabled only if the area, president and arrival time info are filled in
- Notes Screen: the send note button should be enabled if the user added some text

buttons

complete UI in figma: https://www.figma.com/file/21x1ui3YZJrGnpZNVnEuuQ/DRAFT-MV---mobile

### PR DESCRIPTION
**Pull request by pr0t3us**
1. Added background selectors for the `Button` and `Button.Black` style.
2. Added `ViewsValidationButton` which should validate the fields based on the set validators (via `OnChangeValidator` interface). 
3. Created several validators to handle different view types: `RadioGroupsValidator`, `SpinnersValidator` and `TextViewsValidator`.
4. Used `RadioGroupsValidator` and `TextViewsValidator` to handle views in `BranchDetailsFragment`, `SpinnersValidator` and `TextViewsValidator` in `BranchSelectionFragment` and `TextViewsValidator` in `NoteFragment`

Fixes #47

### CODE REVIEW

File: app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginActivity.kt [hunk 1]
```diff
@@ -11,6 +11,7 @@ import ro.code4.monitorizarevot.helper.startActivityWithoutTrace
 import ro.code4.monitorizarevot.ui.base.BaseActivity
 import ro.code4.monitorizarevot.ui.branch.BranchActivity
 import ro.code4.monitorizarevot.ui.onboarding.OnboardingActivity
+import ro.code4.monitorizarevot.widget.validation.TextViewsValidator
```
Comment (AlexandraDaraban) [inline app/src/main/java/ro/code4/monitorizarevot/ui/login/LoginActivity.kt:14]: Remove unused import

File: app/src/main/res/drawable/bg_button_dark.xml [hunk 2]
```diff
@@ -1,8 +0,0 @@
-<?xml version="1.0" encoding="utf-8"?>
-<shape xmlns:android="http://schemas.android.com/apk/res/android"
-    android:padding="10dp"
-    android:shape="rectangle">
-    <!-- you can use any color you want I used here gray color-->
-    <solid android:color="?attr/colorAccent" />
-    <corners android:radius="4dp" />
-</shape>
```
Comment (AlexandraDaraban) [inline app/src/main/res/drawable/bg_button_dark.xml:8]: Delete resource if not used

File: app/src/main/java/ro/code4/monitorizarevot/ui/branch/details/BranchDetailsFragment.kt [hunk 3]
```diff
@@ -81,12 +80,17 @@ class BranchDetailsFragment : BaseFragment<BranchViewModel>() {
     }
 
     private fun setContinueButton() {
+        val radioGroupsValidator = RadioGroupsValidator(
+            environmentRadioGroup,
+            genderRadioGroup
+        )
+        val textViewsValidator = TextViewsValidator(arrivalTime, departureTime)
```
Comment (AlexandraDaraban) [inline app/src/main/java/ro/code4/monitorizarevot/ui/branch/details/BranchDetailsFragment.kt]: Departure time is optional and should be removed from here

**Review by AlexandraDaraban — CHANGES_REQUESTED** [review]
Hey, @pr0t3us. Thank you for your contribution. The PR looks good, however there are some issues that I would like you to address before merging the PR. Also, I cloned your repo locally and I noticed the text of the buttons doesn't change when the button is disabled. Could you look more into this, please? Thanks 😄

**Comment by pr0t3us** [comment]
Hi @AlexandraDaraban thank you for your fair review. I've addressed your comments. Looks like I've imported the wrong alpha channel value for the disabled button color, also I suggest using a more pronounced color for the normal state.

**Comment by AlexandraDaraban** [comment]
Closed in #115

**Pull request closed without merging**